index.js

```diff
@@ -54,7 +55,9 @@ module.exports = (opts) => {
       // proxy should ignore message body when it's a GET or HEAD request
       // when proxy this request, we should reset the content-length to make it a valid http request
       if (req.method === 'GET' || req.method === 'HEAD') {
-        headers['content-length'] = 0
+        if (headers['content-length']) {
```

Comment:
Hi @yohayg, so your point here is to fully ignore the `content-length` header?
Why not just `delete headers['content-length']`, there is no need to iterate over all available headers. Or I am missing something?

Reply:
delete is not recommended since it is relatively slow: https://stackoverflow.com/questions/43594092/slow-delete-of-object-properties-in-js-in-v8/44008788
Iteration on a small number headers would be faster.
I'll check tomorrow if setting the value to undefined is sufficient

Reply:
I guess the the delete is ok as long as it wrapped with the if:

```
        if (headers['content-length']) {
           delete headers['content-length']
       }
```

Reply:
> delete is not recommended since it is relatively slow: https://stackoverflow.com/questions/43594092/slow-delete-of-object-properties-in-js-in-v8/44008788
> Iteration on a small number headers would be faster.
> I'll check tomorrow if setting the value to undefined is sufficient

Hi @yohayg, although it is hard to believe, you are totally right regarding `delete obj...` performance. I have ran some performance tests on my side, and the results looks consistent:
```
[
  { name: 'filter', mean: 1.6885332976677362e-7 },
  { name: 'delete', mean: 3.132603825716916e-7 },
  { name: 'delete if present', mean: 3.358151192450226e-7 }
]
```

So we should go with your proposal in this case. 👏

Reply:
Wouldn't it be enough to set it to `undefined`? I don't think it gets included in that case.

Reply:
Setting it to `undefined` breaks other parts of the solution, as the key has to be excluded anyway.